OpenFileIO.md

```diff
@@ -0,0 +1,86 @@
+# "OpenFileIO" Proposal
+
+## Abstract
+
+This proposal calls for the creation of a minimal, open source C library
+based on the prior art of the OpenImageIO `IOProxy`[^1] and Houdini File
+System[^2] APIs. Providing a common cross-platform, cross-application
+extensible abstraction of a random access byte stream. Its purpose is
+to aid the migration away from `fopen` et. al. as the de-facto low-level
+data access mechanism within digital content creation tooling and
+infrastructure. This is required to allow secure data access via other
+means without the need for complex user-space file system emulations or
+elaborate data orchestration techniques.
+
+> N.B. "OpenFileIO" is used colloquially as a place-holder name.
+
+## Problem Statement
+
+It is currently extremely difficult to provide data access to processing
+operations in a media production pipelines though anything other than
+the standard filesystem. This hampers industry adoption of modern
+storage, security and compute architectures.
+
+## Background
+
+Within the Media and Entertainment sector Digital Content Creation (DCC)
+tools require data to be read, processed and written as part of their
+operation. Traditionally data is accessed via the standard filesystem
+system calls, often targeting a network mount of a shared storage
+resource. With the advent of modern distributed systems architectures,
+there is a strong desire to use other mechanism for the secure
+transport and storage of necessary data.
```

Comment:
You mention secure transport a couple of times. Is this a current explicit problem, and should it be mentioned as a goal?

Reply:
Expanded the goal [here](https://github.com/foundrytom/proposals/pull/1/commits/b69223b58c216c5f4cb893ee068e4af7a62d5763).